api/api_test.go

```diff
@@ -80,7 +81,7 @@ func newTestInstanceWithProfileFromNode(ctx context.Context, node *p2p.QriNode)
 func newTestNodeWithNumDatasets(t *testing.T, _ int) (node *p2p.QriNode, teardown func()) {
 	var r repo.Repo
 	r, teardown = newTestRepo(t)
-	node, err := p2p.NewQriNode(r, config.DefaultP2PForTesting())
+	node, err := p2p.NewQriNode(r, config.DefaultP2PForTesting(), &event.NilBus)
```

Comment:
no need to take a pointer to the nil bus, it's a an exported variable.
```suggestion
	node, err := p2p.NewQriNode(r, config.DefaultP2PForTesting(), event.NilBus)
``

That change will need to made throughout this PR